### DISCUSSION TITLE
Variable size for Header and Footer images

### DISCUSSION THREAD

**Discussion by XiaoKe89** [TheEpicSnowWolf/Foundry-VTT-Prettier-Login-Screen#14] [Questions]
Is there a way to have the size of the header and footer images be variable depending on screen resolution? I have them set at a fixed size but the issue is they look fine on one resolution but tiny in higher resolutions, I was hoping I could get the images to scale depending on the monitor resolution as opposed to just scaling based on one resolution size

**Comment by dandln** [comment]
Hm, technically both images are scaled to fit the other UI elements per default (i.e. the width of the header image is contained to the width of the login form itself), so I'm surprised that the images would be small on higher resolutions while the rest is not? Unless the images themselves are really low res. I've got a fairly high resolution as well and can't reproduce the issue. How high are we talking? Do you have some screenshots maybe?

That being said, you could of course apply other scales to the scale definitions in the CSS, e.g. using % instead of px, but I doubt that this will give you the result you want.

**Reply by XiaoKe89** [reply]
So it looks like the login ui is also resizing it's just less noticeable. After checking the resolution on the images is lower so let me see if I can get higher resolution images and see if that helps, I originally set it up on a lower resolution monitor so maybe I just didn't realize the images were lower resolution to begin with

**Reply by XiaoKe89** [reply]
Hmm doesn't look like that did anything, it's still shrinking the ui to pretty small on the higher resolution monitor. The lower res monitor is 1920x1080 and the higher one is 2560x1440

Here's a couple of screenshots to show the difference
![Screenshot 2023-04-11 013347](https://user-images.githubusercontent.com/86465316/231096909-416fc9f1-4e4e-4ede-bd18-14109e12fcbe.png)
![Screenshot 2023-04-11 013423](https://user-images.githubusercontent.com/86465316/231096915-fc6e244e-1a4e-497f-b7df-d126bc627b08.png)

**Reply by dandln** [reply]
Is this on the same pc/laptop? I believe it might have something to do with how your system is scaling UI elements to the resolution. Just for comparison, this is how my Foundry instance looks like on my MacBook with a 2560x1664 display, notably the login mask looks bigger than in both your screenshots. :D
<img width="1469" alt="image" src="https://user-images.githubusercontent.com/6506715/232338850-7cafe0df-e1e5-4ecf-b01e-5d5fba6d69e2.png">

Basically, the OS can determine how UI elements are scaled on higher-res screens to basically work against exactly the issue that you're having here. You can adjust that behaviour under Start > Settings > System > Display on Windows, and under System Settings > Displays on Mac. I wonder, are all websites appearing small(er) on your 1440p screen?

**Reply by XiaoKe89** [reply]
Well I tried it on my laptop and it looks like the first screenshot and I also have a 1080p monitor on the same machine as the 1440p monitor and it looks like the top screenshot on the 1080p monitor. My scale on my PC is set to %100 and all other sites seem to be scaling fine nothing else seems to be too much smaller

**Reply by dandln** [reply]
Fair enough. Well, not sure if this actually solves your problem, but you could play around with the dimensions of the different elements.

## Changing the login mask's width
As I said, the world logo's maximum width is bound to the width of the login mask, which you can change in this definition:
``` css
/* Login mask width and position */
#join-game {
  width: 400px !important;
  top: calc(50% - 300px) !important;
}
```

You could change `400px` to a wider, fixed value, and then adjust the height of the logo image here:
``` css
/* World Logo - REMOVE THIS IF YOU WANT A TEXT TITLE */
#join-game #world-title {
  (...)
  height: 120px !important;
  (...)
}
```

You will then also need to play around with the `top` value in the `#join-game` defintion to re-center the login mask vertically. Note that you could theoretically also use a percentile value for the `#join-game` width to scale the mask dynamically with the width of the browser window, however this would make the logo eventually glitch out on smaller screens if it's set to a fixed height.

## Make the logo "break out" of the login mask
The other thing you could do is make the logo itself bigger, outside the boundries of the login mask. To do that, you will want to widen the container for the logo by adding the following properties to the world logo definition:
``` css
#join-game #world-title {
  ...
  width: 400%;
  margin-left: -150% !important;
}
```

This will make the container *very* wide and you can then make the logo bigger using the `height` porperty in the world logo definition again. You could change the values here to adjust your needs, The `width` property sets the width of the image container, and a negative `margin-left` property will move it horizontically, making sure that it stays centred.

---

Neither of these things really make your UI elements bigger depending on the screen resolution. You could theoretically copy and implement different definitions of the stylesheet depending on the screen size using CSS [media queries](https://www.w3schools.com/css/css3_mediaqueries_ex.asp), but again, this really shouldn't be an issue so I assume there's something different going on in your browser causing the UI elements to appear very small that has little to do with the CSS of this module.
